icu4c/source/common/stringpiece.cpp

```diff
@@ -51,6 +51,47 @@ void StringPiece::set(const char* str) {
     length_ = 0;
 }
 
+int32_t StringPiece::find(StringPiece needle, int32_t offset) {
+  if (length() == 0 && needle.length() == 0) {
+    return 0;
+  }
+  // TODO: Improve to be better than O(N^2)?
```

Comment:
I looked up in a reference book to see what it said.  For sufficiently short strings, the O(mn) algorithm is hard to beat in practice.  For lengths < 5, it's near impossible.  Only for sufficiently long strings does the Knuth-Morris-Pratt algorithm of intelligently advancing i make a difference (https://en.wikipedia.org/wiki/Knuth%E2%80%93Morris%E2%80%93Pratt_algorithm)

Reply:
Acknowledged